cmd/cored/main.go

```diff
@@ -143,6 +147,55 @@ func runServer() {
 	ctx := context.Background()
 	env.Parse()
 
+	raftDir := filepath.Join(*dir, "raft") // TODO(kr): better name for this
+	raftDB, err := raft.Start(*listenAddr, raftDir, *bootURL)
+	if err != nil {
+		chainlog.Fatal(ctx, chainlog.KeyError, err)
+	}
+
+	mux := http.NewServeMux()
+	mux.Handle("/raft/", raftDB)
+
+	var handler http.Handler = mux
+	handler = reqid.Handler(handler)
+
+	secureheader.DefaultConfig.PermitClearLoopback = true
+	secureheader.DefaultConfig.HTTPSRedirect = httpsRedirect
+	secureheader.DefaultConfig.Next = handler
+
+	server := &http.Server{
+		Addr:         *listenAddr,
+		Handler:      secureheader.DefaultConfig,
+		ReadTimeout:  httpReadTimeout,
+		WriteTimeout: httpWriteTimeout,
+		// Disable HTTP/2 for now until the Go implementation is more stable.
+		// https://github.com/golang/go/issues/16450
+		// https://github.com/golang/go/issues/17071
+		TLSNextProto: map[string]func(*http.Server, *tls.Conn, http.Handler){},
+	}
+
+	go func() {
```

Comment:
1. It moved up because we need to start listening on the raft routes (`/raft/`) before we can set up the rest of the server (e.g. check if there's a config). We can't even read from raft storage without making this new node available to raft, because raft requires a round of consensus for reads as well as writes. 

2. It sprouted a goroutine (🌱 ) because the `ListenAndServe` calls are blocking and we needed to proceed through the rest of the set up. 

I'll add comments explaining all of this.